src/users/users.service.ts

```diff
@@ -0,0 +1,166 @@
+    const jwtToken = this.tokenService.generateToken(user.user_id);
+    const expiredTime = this.tokenService.EXPIRE_TIME;
+
+    return { user: saveUser, access_token: jwtToken, expiredTime };
+  }
+
+  async login(loginUserDto: LoginUserDto): Promise<LoginSuccessDto> {
+    const { email, password } = loginUserDto;
+
+    const user = await this.userRepository.findOne({ where: { email } });
+    if (!user) throw new UnauthorizedException();
+    const login = await this.loginRepository.findOne({
+      where: { user_id: user.user_id },
+    });
+    const isValid = await login.comparePassword(password);
+    if (!isValid) throw new UnauthorizedException();
+
+    const jwtToken = this.tokenService.generateToken(user.user_id);
+    const expiredTime = this.tokenService.EXPIRE_TIME;
+    return { user, access_token: jwtToken, expiredTime };
+  }
+
+  async findUserById(user_id: string): Promise<object> {
+    const userInfo = await this.userRepository.findOne({
+      where: { user_id },
+      relations: ['tags'],
+    });
+    if (!userInfo)
+      throw new HttpException('user id not found', HttpStatus.BAD_REQUEST);
+
+    return userInfo;
+  }
+
+  async updateUserProfile(body: UserProfileDto, file: object): Promise<object> {
+    const {
+      user_id,
+      name,
+      email,
+      birth,
+      location,
+      sick_year,
+      hospital,
+      level,
+      carer,
+      curr_problem,
+      tags,
+    } = body;
+
+    let { picture } = body;
+
+    const userInfo = await this.userRepository.findOne({
+      where: { user_id },
+    });
+    if (!userInfo)
+      throw new HttpException('user id not found', HttpStatus.BAD_REQUEST);
+
+    // 加入註解
+    if (file) {
+      const fileResponse = await this.s3Service.uploadFile(file);
+      picture = fileResponse.Location;
+      userInfo.picture = picture;
+    }
+
+    if (!file && picture) {
+      userInfo.picture = picture;
+    }
+
+    userInfo.name = name;
+    userInfo.email = email;
+    userInfo.birth = birth;
+    userInfo.location = location;
+    userInfo.sick_year = sick_year;
+    userInfo.hospital = hospital;
+    userInfo.level = level;
+    userInfo.carer = carer;
+    userInfo.curr_problem = curr_problem;
+    const userInfoSave = await this.userRepository.save(userInfo);
+    await this.saveUserTags(userInfo.user_id, tags);
+    return userInfoSave;
+  }
+
+  async saveUserTags(user_id: string, tagIds: any): Promise<object> {
+    if (tagIds && tagIds.length > 0) {
+      const userToUpdate = await this.userRepository.findOne({
+        where: { user_id },
+      });
+
+      if (userToUpdate) {
+        const newTagsObj = await this.tagRepository.find({
+          where: { tag_id: In(tagIds) },
+        });
+
+        userToUpdate.tags = newTagsObj;
+        const result = await this.userRepository.save(userToUpdate);
+        return result;
+      }
+    }
+  }
+
+  async findUserTags(user_id: string): Promise<object> {
```

Comment:
Have another PR to correct all your typings. 
For example, `Promise<object>` should be `Promise<Tag>` in this example.

Reply:
ok